apps/web/src/services/dashboards/shared/dashboard-variables/DashboardVariablesDropdown.vue

```diff
@@ -180,7 +180,8 @@ const loadSearchResourceOptions = async () => {
 
 const initVariable = () => {
     const variableSchema = dashboardDetailState.variablesSchema.properties[props.propertyName];
-    if ((variableSchema.options as SearchResourceOptions)?.type === 'SEARCH_RESOURCE') {
+    const existingVariableInfoValue = dashboardDetailState.dashboardInfo?.variables[props.propertyName];
+    if ((variableSchema.options as SearchResourceOptions)?.type === 'SEARCH_RESOURCE' && !existingVariableInfoValue) {
```

Comment:
suggestion: I think it is more safe to check if the value is not undefined because variable can be '0' or 'false' or whatever that can be coverted as `false`.